mermithid/processors/misc/TritiumAndEfficiencyBinner.py

```diff
@@ -0,0 +1,136 @@
+
+class TritiumAndEfficiencyBinner(BaseProcessor):
+
+    def InternalConfigure(self, params):
+        '''
+        Configure
+        '''
+
+        # Read other parameters
+        self.namedata = reader.read_param(params, 'variables', "required")
+        self.N = reader.read_param(params, 'N', 'N')
+        self.eff_eqn = reader.read_param(params, 'efficiency', '1')
+        self.bins = reader.read_param(params, 'bins', [])
+        self.asInteger = reader.read_param(params, 'asInteger', False)
+        self.energy_or_frequency = reader.read_param(params, 'energy_or_frequency', 'energy')
+        self.efficiency_filepath = reader.read_param(params, 'efficiency_filepath', '/host/combined_energy_corrected_eff_at_quad_trap_frequencies.json')
+        self.fss_bins = reader.read_param(params, "fss_bins", False)
+        # If self.fss_bins is True, self.bins is ignored and overwritten
+
+        # initialize the histogram to store the corrected data
+        if self.energy_or_frequency == 'energy':
+            print(sys.getrefcount(self.corrected_data))
+            self.output_bin_variable='KE'
+        elif self.energy_or_frequency == 'frequency':
+            self.output_bin_variable='F'
+
+        if self.fss_bins == True:
+            a = self.GetEfficiencyFileContent()
+            self.bin_centers = a['frequencies']
+            self.bins = np.array(self.bin_centers) - (self.bin_centers[1]-self.bin_centers[0])/2
+            self.bins = np.append(self.bins, [self.bin_centers[-1]+(self.bin_centers[1]-self.bin_centers[0])/2])
+        else:
+            self.bin_centers = self.bins[0:-1]+0.5*(self.bins[1]-self.bins[0])
+        return True
+
+    def InternalRun(self):
+        print('namedata:',self.namedata)
+
+        N,b = np.histogram(self.data[self.namedata], self.bins)
+
+        # Efficiencies for bins
+        # If we want to use fss bins, we want False to be passed to EfficiencyAssignment
+        # because we just want to use the efficiency info directly from the file.
+        self.bin_efficiencies, self.bin_efficiency_errors = self.EfficiencyAssignment(self.bin_centers, self.bins, not self.fss_bins)
+        self.bin_efficiencies_normed = self.bin_efficiencies/np.sum(self.bin_efficiencies)
+        self.bin_efficiency_errors_normed = self.bin_efficiency_errors/np.sum(self.bin_efficiencies)
+
+        # Efficiencies for events, but with binned uncertainties (?)
+        self.event_efficiencies, self.event_efficiency_errors = self.EfficiencyAssignment(self.data[self.namedata], None, False)
+        self.event_efficiencies_normed = self.event_efficiencies/np.sum(self.event_efficiencies)
+        self.event_efficiency_errors_normed = self.event_efficiency_errors/np.sum(self.event_efficiencies)
+
+        # Put it all in a dictionary
+        temp_dictionary = {self.output_bin_variable: [], 'N': [], 'bin_efficiencies': [], 'bin_efficiency_errors': [], 'event_efficiencies': [], 'event_efficiency_errors': []}
```

Comment:
why using a temporary dictionary instead of putting everything in the results one directly?